CefSharp/Internals/JavascriptObjectRepository.cs

```diff
@@ -104,15 +104,33 @@ public bool TryCallMethod(long objectId, string name, object[] parameters, out o
 
             try
             {
-                // Do we have enough arguments? Add Type.Missing for any that we don't have incase of optional params
+                //Fix for C# paramArray and refactor on arguments mismatch 03/22/2016
                 var missingParams = method.ParameterCount - parameters.Length;
-                if (missingParams > 0)
+                if (missingParams > 0 || method.MethodParameters.Any(t => t.IsParamArray))
                 {
-                    var paramList = new List<object>(parameters);
+                    var paramList = new List<object>();
 
-                    for (var i = 0; i < missingParams; i++)
+                    for (int i = 0; i < method.MethodParameters.Count; i++)
                     {
-                        paramList.Add(Type.Missing);
+                        object jsParam = parameters.ElementAtOrDefault(i);
+                        if (jsParam == null && !method.MethodParameters[i].IsParamArray)
+                        {
+                            paramList.Add(Type.Missing);
+                        }
+                        else if (method.MethodParameters[i].IsParamArray)
+                        {
+                            List<object> convertedParams = new List<object>();
+                            for (int s = i; s < parameters.Count(); s++)
```

Comment:
My mistake, should use the `Count`, property, there's no need to call the method (which is an `Extension` method from memory, has a small overhead).